### PR TITLE
[Merged by Bors] - TY-3022 update rust toolchain 1.62.1

### PR DESCRIPTION
**Pull request by janpetschexain**
**References**

- [TY-3022]
- requires https://github.com/xaynetwork/xayn_docker/pull/16

**Summary**

- update rust toolchain to `1.62.1`/`nightly-2022-07-16`
- update just to `1.3.0`
- increment docker ci tag
- fix clippy lints
- update rust dependencies & subdependencies

[TY-3022]: https://xainag.atlassian.net/browse/TY-3022?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: discovery_engine_core/core/Cargo.toml [hunk 1]
```diff
@@ -39,11 +39,10 @@ sqlx = { version = "0.6.0", features = ["runtime-tokio-rustls", "sqlite", "uuid"
 
 [dev-dependencies]
 async-once-cell = "0.4.2"
-claim = "0.5.0"
```
Comment (janpetschexain) [inline discovery_engine_core/core/Cargo.toml:42]: for whatever reason `claim` fixes a subdependency to an old minor version, which results in version conflicts for example with `tokio`. i removed `claim` and replaced its macros with `std` macros.

File: justfile [hunk 2]
```diff
@@ -199,8 +200,14 @@ rust-clean:
     cargo clean
 
 # Cleans up darts build cache
+_dart-clean $WORKSPACE:
+    cd "$WORKSPACE"; \
+    find . -type d -name .dart_tool -prune -exec rm -r '{}' \;
```
Comment (janpetschexain) [inline justfile:205]: this exits with an error on macos if the `.` is not given as a path.

Comment (Robert-Steiner) [reply justfile:205]: @rustonaut can maybe you check if this still works for you?

File: .github/actions/setup-job-docker/action.yml [hunk 3]
```diff
@@ -23,6 +23,9 @@ runs:
     - name: Load .env
       uses: xom9ikk/dotenv@d3ff95524814ceac377510f30f4af6296ea612c1 #v1.0.2
 
+    - name: Setup cmake
+      uses: jwlawson/actions-setup-cmake@884cf5c591490f86b788d68b6f585f26235578d6 # v1.12
```
[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply .github/actions/setup-job-docker/action.yml:27]: looks like that they removed it again in the next version https://github.com/tokio-rs/prost/issues/646

**Review by Robert-Steiner — APPROVED** [review]
lgtm
we could also update the sub deps with `cargo update`

File: justfile (hunk 2 above)
Comment (janpetschexain) [reply justfile:205]: it should work as this is the same syntax used for example in `just remove-assets`

**Comment by janpetschexain** [comment]
bors merge

**Pull request closed without merging**